sdk/src/main/jni/utils/serializer.c

```diff
@@ -23,15 +25,29 @@ bsg_report_header *bsg_report_header_read(int fd);
 }
 #endif
 
+char report_path[384];
+
 bool bsg_serialize_report_to_file(bsg_environment *env) {
-  int fd = open(env->next_report_path, O_WRONLY | O_CREAT, 0644);
+  // copy path to variable so that false positives can be deleted after the JNI env
+  // has been deallocated
+  bsg_strcpy(report_path, env->next_report_path);
+
+  int fd = open(report_path, O_WRONLY | O_CREAT, 0644);
   if (fd == -1) {
     return false;
   }
-
   return bsg_report_write(&env->report_header, &env->next_report, fd);
 }
 
+bool bsg_delete_report_file() {
```

Comment:
Why isn't the file path passed as an argument from `bsg_handle_signal`? The copy could happen prior to uninstalling the handler.